src/postgraphile-upsert.ts

```diff
@@ -354,20 +423,26 @@ function createUpsertField({
                 sqlValues.push(
                   gql2pg(whereClauseValue, attr.type, attr.typeModifier)
                 );
+              } else if (setTimestamp[attr.name]) {
+                conflictOnlyColumns.push(sql.identifier(attr.name))
               }
             });
 
             // Construct a array in case we need to do an update on conflict
-            const conflictUpdateArray = sqlColumns.map(
+            const conflictUpdateArray = conflictOnlyColumns.concat(sqlColumns).filter((col) => ignoreUpdate[col.names[0]] != true).map(
               (col) =>
                 sql.query`${sql.identifier(
                   col.names[0]
-                )} = excluded.${sql.identifier(col.names[0])}`
+                )} = ${setTimestamp[col.names[0]] ? sql.fragment`CURRENT_TIMESTAMP` : sql.fragment`excluded.${sql.identifier(col.names[0])}`}`
             );
             assert(table.namespace, "expected table namespace");
 
             // SQL query for upsert mutations
             // see: http://www.postgresqltutorial.com/postgresql-upsert/
+            const conflictAction = conflictUpdateArray.length == 0 ? sql.fragment`do nothing` : sql.fragment`on constraint ${sql.identifier(
+              constraintName
+            )}
+            do update set ${sql.join(conflictUpdateArray, ", ")}`
```

Comment:
All of this processing for these new fields _I believe_ should probably require activation from a flag to enable the functionality. e.g. `createPlugin({ enableQueryDefinedConflictResolutionTuning: boolean })`

We may not want this functionality activated out-of-the box--i think users need to opt-in. The plugin allows a pretty simple mental model right now, but this feature let's our users' users control maybe more than some teams would like.